### PR TITLE
dcos-net: log resolv.conf diff

### PR DESCRIPTION
**Pull request by jkoelker**
## High-level description

When replacing the resolv.conf instead of just showing the new
resolv.conf, show the diff of the modification.

## Corresponding DC/OS tickets (required)

  - [COPS-6411](https://jira.d2iq.com/browse/COPS-6411) Hostname Resolution fails for VIPs.

### CODE REVIEW

**Comment by jkoelker** [comment]
I couldn't find any testing already for this script, so I just manually tested it, by editing the `resolv.conf` on node running this branch:

```
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal systemd[1]: Starting Generate resolv.conf: configures network name resolution...
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: Updating /etc/resolv.conf
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: # Generated by gen_resolvconf.py. Do not edit.
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: # Change configuration options by changing DC/OS cluster configuration.
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: # This file must be overwritten regularly for proper cluster operation around
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: # master failure.
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: options timeout:1
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: options attempts:3
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: nameserver 198.51.100.1
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: nameserver 198.51.100.2
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: nameserver 198.51.100.3
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: --- /etc/resolv.conf
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: +++ /etc/resolv.conf.tmp
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: @@ -6,5 +6,6 @@
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: options timeout:1
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: options attempts:3
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: -nameserver 8.8.8.8
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: -nameserver 8.8.4.4
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: +nameserver 198.51.100.1
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: +nameserver 198.51.100.2
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal gen_resolvconf.py[17889]: +nameserver 198.51.100.3
Aug 19 01:07:51 ip-172-16-15-242.us-west-2.compute.internal systemd[1]: Started Generate resolv.conf: configures network name resolution.
```

**Comment by jkoelker** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by jkoelker** [comment]
@mesosphere-mergebot request-review @vinodkone @jongiddy @janisz

File: packages/dcos-net/extra/gen_resolvconf.py [hunk 1]
```diff
@@ -37,7 +38,7 @@ def check_server(addr):
     except dns.exception.Timeout:
         print('Skipping DNS server {}: no response'.format(
             addr), file=sys.stderr)
-    except:
+    except Exception:
         print("Unexpected error querying DNS for server \"{}\" exception: {}".format(
             addr, sys.exc_info()[1]))
```
Comment (janisz) [inline packages/dcos-net/extra/gen_resolvconf.py:43]: I know it's not a part of diff but is this a proper way of printing exceptions?

**Review by janisz — APPROVED** [review]

File: packages/dcos-net/extra/gen_resolvconf.py (hunk 1 above)
Comment (jkoelker) [reply packages/dcos-net/extra/gen_resolvconf.py:43]: Yea not really, doing:

```python
except Exception as e:
    print("Unexpected error querying DNS for server \"{}\" exception: {}".format(addr, e))
```

Would be the right way™. After looking into this script, ugh, it should be heavily rewritten. I was just fixing the flake8 linting error of bare except ;).

**Comment by jkoelker** [comment]
@mesosphere-mergebot sync

**Comment by jkoelker** [comment]
@mesosphere-mergebot sync

**Comment by jkoelker** [comment]
@mesosphere-mergebot sync

**Comment by jkoelker** [comment]
@mesosphere-mergebot backport 1.13 2.0 2.1

**Pull request merged**